streamparse/bolt.py

```diff
@@ -216,12 +216,16 @@ def run(self):
             self.initialize(storm_conf, context)
             while True:
                 self._current_tups = [read_tuple()]
-                self.process(self._current_tups[0])
-                if self.auto_ack:
-                    self.ack(self._current_tups[0])
-                # reset so that we don't accidentally fail the wrong tuples
-                # if a successive call to read_tuple fails
-                self._current_tups = []
+                tup = self._current_tups[0]
+                if tup.task == -1 and tup.stream == '__heartbeat':
+                    self.send_message({'command': 'sync'})
```

Comment:
Ah, good catch.  In my Perl implementation the IPC things are all methods of the `Component` base class.